pallene/ir.lua

```diff
@@ -45,11 +45,13 @@ end
 
 function ir.Function(loc, name, typ)
     return {
-        loc = loc,           -- Location
-        name = name,         -- string
-        typ = typ,           -- Type
-        vars = {},           -- list of ir.VarDecl
-        body = false,        -- ir.Cmd
+        loc = loc,          -- Location
+        name = name,        -- string
+        typ = typ,          -- Type
+        vars = {},          -- list of ir.VarDecl
+        captured_vars = {}, -- { integer => ast.Decl }
+        id_of_upval = {},   -- { integer => integer }
```

Comment:
Let's use a separatge list of upvalues, instead of reserving space in the local variables